### PR TITLE
storageos-operator: Sync with rancher chart

### PR DESCRIPTION
**Pull request by darkowlzz**
Updates `storageos/storageos-operator` chart to be the same as `rancher/storageos-operator` chart.

### CODE REVIEW

**Comment by darkowlzz** [comment]
Added changes to update to operator 1.2.0 as well.

File: stable/storageos-operator/templates/cleanup.yaml [hunk 1]
```diff
@@ -114,14 +114,17 @@ spec:
...
         image: bitnami/kubectl:1.14.1
         command:
           - kubectl
+          - -n
+          - {{ $.Values.cluster.namespace }}
           - delete
           {{- range .command }}
           - {{ . | quote }}
           {{- end }}
+          - --ignore-not-found=true
```
Comment (darkowlzz) [inline stable/storageos-operator/templates/cleanup.yaml]: Avoid failure if the resource is not found.

Comment (croomes) [reply stable/storageos-operator/templates/cleanup.yaml]: :+1: Does this need to be in the next Rancher PR?

Comment (darkowlzz) [reply stable/storageos-operator/templates/cleanup.yaml]: Yes, it'll also help not fail when we try to delete CSI deployment/statefulset and one of them is not found. We need this.

Comment (croomes) [reply stable/storageos-operator/templates/cleanup.yaml]: added in Rancher PR

File: stable/storageos-operator/README.md [hunk 2]
```diff
@@ -148,9 +148,9 @@ Operator chart and their default values.
...
 --------- | ----------- | -------
-`operator.image.repository` | StorageOSCluster container image repository | `storageos/cluster-operator`
-`operator.image.tag` | StorageOSCluster container image tag | `1.1.0`
-`operator.image.pullPolicy` | StorageOSCluster container image pull policy | `IfNotPresent`
+`operator.image.repository` | StorageOS Operator container image repository | `storageos/cluster-operator`
+`operator.image.tag` | StorageOS Operator container image tag | `1.2.0`
+`operator.image.pullPolicy` | StorageOS Operator container image pull policy | `IfNotPresent`
 `podSecurityPolicy.enabled` | If true, create & use PodSecurityPolicy resources | `false`
 `podSecurityPolicy.annotations` | Specify pod annotations in the pod security policy | `{}`
 `cluster.create` | If true, auto-create the StorageOS cluster | `true`
```
Comment (croomes) [inline stable/storageos-operator/README.md]: Can you add a doc line for `cluster.csi.deploymentStrategy`?  I did it at the end of the block:  
https://github.com/rancher/charts/pull/163/files#diff-e72916d3373dc9723fce013bd3357dbcR176

File: stable/storageos-operator/values.yaml [hunk 3]
```diff
@@ -88,10 +88,11 @@ cluster:
     # [Docker Hub](https://hub.docker.com/r/storageos/node/).
     node:
       repository: storageos/node
-      tag: 1.2.0
+      tag: 1.2.1
 
   csi:
     enable: true
+    deploymentStrategy: statefulset
```
Comment (croomes) [inline stable/storageos-operator/values.yaml]: This can change to `deployment`

**Pull request merged**